File: extras/keypad.java
import java.util.*;

public class keypad {
    public static Scanner scn = new Scanner(System.in);

    public static void main(String[] args) {
        System.out.println(keypad_01("11024", ""));
        // System.out.println(keypad_01("234", ""));
        // System.out.println(keypad_02("234"));
        System.out.println(keypad_02("11024"));
    }

    static String words[] = { "!@$", "abc", "def", "ghi", "jkl", "mno", "pqrs", "uvw", "xyz", "*", "#", "%^&" };

    public static int keypad_01(String str, String ans) {

        if (str.length() == 0) {
            System.out.println(ans);
            return 1;
        }
        int count = 0;
        int idx = str.charAt(0) - '0';
        String roq = str.substring(1);
        String word = words[idx];

        for (int i = 0; i < word.length(); i++) {
            count += keypad_01(roq, ans + word.charAt(i));
        }

        if (str.length() >= 2) {
            idx = idx * 10 + (str.charAt(1) - '0');
            if (idx >= 10 && idx <= 11) {
                word = words[idx];
                for (int i = 0; i < word.length(); i++) {
                    count += keypad_01(str.substring(2), ans + word.charAt(i));
                }
            }
        }
        return count;
    }

    public static ArrayList<String> keypad_02(String str){
        if (str.length() == 0){
            ArrayList<String> base = new ArrayList<>();
            base.add("");
            return base;
        }

        int idx = str.charAt(0) - '0';
        String roq = str.substring(1);
        String word = words[idx];

        ArrayList<String> recAns = keypad_02(roq);
        
        ArrayList<String> myAns = new ArrayList<>();
        for (int i = 0; i < word.length(); i++){
            for (String s : recAns){
                myAns.add(word.charAt(i) + s);
            }
        }

        if (str.length() > 1){
            idx = idx * 10 + (str.charAt(1) - '0');
            if (idx == 10 || idx == 11){
                word = words[idx];
                recAns = keypad_02(str.substring(2));   
                for (int i = 0; i < word.length(); i++){
                    for (String s : recAns){
                        myAns.add(word.charAt(i) + s);
                    }
                }
            }
        }
        return myAns;
    }
}

File: extras/keypad.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

string words[] = {"!@$", "abc", "def", "ghi", "jkl", "mno", "pqrs", "uvw", "xyz", "*", "#", "%^&"};
int keypad_01(string str, string ans)
{
    if (str.size() == 0)
    {
        cout << ans << endl;
        return 1;
    }
    int count = 0;
    int idx = str[0] - '0';
    string roq = str.substr(1);
    string word = words[idx];

    for (int i = 0; i < word.size(); i++)
    {
        count += keypad_01(roq, ans + word[i]);
    }

    if (str.size() > 1)
    {
        idx = idx * 10 + (str[1] - '0');
        if (idx == 10 || idx == 11)
        {
            word = words[idx];
            for (int i = 0; i < word.size(); i++)
            {
                count += keypad_01(str.substr(2), ans + word[i]);
            }
        }
    }
    return count;
}

vector<string> keypad_02(string str)
{
    if (str.size() == 0)
    {
        vector<string>* base = new vector<>();
        vector->push_back("");
        return base;
    }

    int idx = str[0] - '0';
    string roq = str.substr(1);
    string word = words[idx];

    vector<string> recAns = keypad_02(roq);

    vector<string> myAns = new vector<>();
    for (int i = 0; i < word.size(); i++)
    {
        for (string s : recAns)
        {
            myAns.push_back(word[i] + s);
        }
    }
    
    if (str.size() > 1)
    {
        idx = idx * 10 + (str[1] -  '0');
        if (idx == 10 || idx == 11)
        {
            word = words[idx];
            recAns = keypad_02(str.substr(2));
            for (int i = 0; i < word.size(); i++)
            {
                for (string s : recAns)
                {
                    myAns.push_back(str[i] + s);
                }
            }
        }
    }
    return myAns;
}

int main(int args, char **argv)
{
    // keypad_01("234", "");
    // cout << keypad_01("11024", "");
    keypad_02("234");
    // keypad_02("11024");
    return 0;
}